x-pack/plugins/enterprise_search/public/applications/workplace_search/views/oauth_authorize/oauth_authorize.tsx

```diff
@@ -0,0 +1,161 @@
+  EuiPage,
+  EuiPageBody,
+  EuiPanel,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiTitle,
+  EuiText,
+  EuiSpacer,
+} from '@elastic/eui';
+
+import { FormattedMessage } from '@kbn/i18n/react';
+
+import { FlashMessages } from '../../../shared/flash_messages';
+import { Loading } from '../../../shared/loading';
+
+import { WORKPLACE_SEARCH_TITLE } from '../../constants';
+
+import {
+  AUTHORIZATION_REQUIRED_TITLE,
+  AUTHORIZE_BUTTON_LABEL,
+  DENY_BUTTON_LABEL,
+  HTTP_REDIRECT_WARNING_MESSAGE,
+  SCOPES_LEAD_IN_MESSAGE,
+  SEARCH_SCOPE_DESCRIPTION,
+  WRITE_SCOPE_DESCRIPTION,
+} from './constants';
+
+import { OAuthAuthorizeLogic } from './oauth_authorize_logic';
+
+export const OAuthAuthorize: React.FC = () => {
+  const { search } = useLocation() as Location;
+  const { initializeOAuthPreAuth, allowOAuthAuthorization, denyOAuthAuthorization } = useActions(
+    OAuthAuthorizeLogic
+  );
+
+  const { buttonLoading, dataLoading, cachedPreAuth } = useValues(OAuthAuthorizeLogic);
+
+  useEffect(() => {
+    initializeOAuthPreAuth(search);
+  }, []);
+
+  if (dataLoading) return <Loading />;
+
+  const showHttpRedirectUriWarning = cachedPreAuth.redirectUri.startsWith('http:');
+
+  const httpRedirectUriWarning = (
+    <>
+      <EuiCallOut title={HTTP_REDIRECT_WARNING_MESSAGE} color="danger" iconType="cross" />
+      <EuiSpacer />
+    </>
+  );
+
+  const scopeDescription = (scopeName: string): string | undefined => {
+    if (scopeName === 'search') {
+      return SEARCH_SCOPE_DESCRIPTION;
+    } else if (scopeName === 'write') {
+      return WRITE_SCOPE_DESCRIPTION;
+    } else {
+      return undefined;
+    }
+  };
+
+  const scopeListItems = cachedPreAuth.scopes.map((scope) => {
+    const scopeDesc = scopeDescription(scope);
+
+    if (scopeDesc) {
+      return <li>{scopeDesc}</li>;
+    } else {
+      return (
+        <li>
+          <FormattedMessage
+            id="xpack.enterpriseSearch.workplaceSearch.oauthAuthorize.UnknownScopeDescription"
+            defaultMessage="{unknownAction} your data"
+            values={{
+              unknownAction: scope,
+            }}
+          />
+        </li>
+      );
+    }
+  });
+
+  return (
+    <EuiPage restrictWidth>
+      <EuiPageBody>
+        <FlashMessages />
+        <EuiSpacer />
+        <EuiPanel paddingSize="l" style={{ maxWidth: 500, margin: '40px auto' }}>
+          <EuiHeaderSection>
+            <EuiHeaderSectionItem>
+              <EuiHeaderLogo iconType="logoWorkplaceSearch" />
+              <EuiTitle>
+                <h1>{WORKPLACE_SEARCH_TITLE}</h1>
+              </EuiTitle>
+            </EuiHeaderSectionItem>
+          </EuiHeaderSection>
+          <EuiSpacer />
+          <EuiTitle>
+            <h2>{AUTHORIZATION_REQUIRED_TITLE}</h2>
+          </EuiTitle>
+          <EuiSpacer />
+          <EuiText>
+            <p>
+              <FormattedMessage
+                id="xpack.enterpriseSearch.workplaceSearch.oauthAuthorize.AuthorizationDescription"
```

Comment:
Can we use camelCase here?
```suggestion
                id="xpack.enterpriseSearch.workplaceSearch.oauthAuthorize.authorizationDescription"
```

Reply:
whoops! I think I introduced that extra dot and missed downcasing the next word.